app/controllers/works_controller.rb

```diff
@@ -0,0 +1,77 @@
+class WorksController < ApplicationController
+  
+  def index
+    @works = Work.all
+  end
+  
+  def show
+    work_id = params[:id]
+    @work = Work.find_by(id: work_id)
```

Comment:
This would be good functionality to put in a controller filter -- a private method in the works controller called something like `find_work`.